### PR TITLE
[DNSAPI][DNSAPI_APITEST] Fix DnsQuery_UTF8 function and add tests

### PR DESCRIPTION
**Pull request by mkonst-reut**
JIRA issue: [CORE-11634](https://jira.reactos.org/browse/CORE-11634)

VirtualBox_ReactOS  2019-08-17 _19_08_2019_01_01_49

### CODE REVIEW

File: dll/win32/dnsapi/dnsapi/query.c [hunk 1]
```diff
@@ -93,6 +93,65 @@ DnsCToW(const CHAR *NarrowString)
...
+    if (WideString == NULL)
+    {
+        return NULL;
+    }
+    MultiByteToWideChar(CP_UTF8,
+                        0,
+                        NarrowString,
+                        -1,
+                        WideString,
+                        WideLen);
```
Comment (ThFabba) [inline dll/win32/dnsapi/dnsapi/query.c]: You multiplied `WideLen` by `sizeof(WCHAR)` above, but the function expects a character count.

File: dll/win32/dnsapi/dnsapi/query.c [hunk 2]
```diff
@@ -906,8 +965,111 @@ DnsQuery_UTF8(LPCSTR Name,
...
+
+    Status = DnsQuery_W(Buffer, Type, Options, Servers, &QueryResultWide, Reserved);
+
+    while (Status == ERROR_SUCCESS && QueryResultWide)
+    {
+        switch (QueryResultWide->wType)
+        {
+        case DNS_TYPE_A:
+        case DNS_TYPE_WKS:
+            ConvertedRecord = RtlAllocateHeap(RtlGetProcessHeap(), 0, sizeof(DNS_RECORD));
```
Comment (ThFabba) [inline dll/win32/dnsapi/dnsapi/query.c]: Check for failure please.

**Comment by learn-more** [comment]
Thanks for your contribution!

It looks like `DnsQuery_A` and `DnsQuery_UTF8` behave exactly the same, except for their helper routine. This helper routine (`DnsWToC`) also behaves the same as `DnsWToUTF8`, except for the codepage.
If the function were to be changed into a generic `DnsQuery_Codepage`, with an additional `CodePage` argument (`CP_ACP` or `CP_UTF8`) the code can be shared between the 2 functions.

File: dll/win32/dnsapi/dnsapi/query.c (hunk 1 above)
Comment (mkonst-reut) [reply dll/win32/dnsapi/dnsapi/query.c]: You are right MultiByteToWideChar function expects a character count
and RtlAllocateHeap function expect size in bytes.
This code was in base version of function DnsQuery_A.
I pushed updates. Please check them.

**Comment by mkonst-reut** [comment]
Created new function DnsQuiry_CodePage on base of DnsQuiry_A with new parameter CodePage.
New functions: DnsQuiry_A and DnsQuiry_UTF8 calls this function with appropriate parameter CodePage.

**Comment by learn-more** [comment]
Thanks!

**Comment by ThFabba** [comment]
Oh... I thought we were also gonna unify DnsWToC and DnsWToUTF8.

**Comment by learn-more** [comment]
That was the idea, yeah

**Pull request merged**